File: Practice/17/py/casino/casino/casino.py
i = 1
max = 0
red = 0
black = 0
noms= []
for temp in range(0,37):
    noms.append(0)
ends = []
for temp in range(0,37):
    ends.append(0)
count = 0
while (i):
    nom = int(input("Введите последний результат\n"))
    if(nom == -1): break
    if (nom >= 0 and nom < 37):
        noms[nom]+=1
        ends[nom] = 1 + count%12
        if (nom % 2 == 0 and nom != 0): black+=1
        if (nom % 2 == 1 and nom != 0): red+=1
        for temp in range(0,37):
            if (noms[temp] > max): max = noms[temp]
        print("Выпадали чаще всего:", end = " ")
        for temp in range(0,37):
            if (noms[temp] == max): print(temp, end = " ")
        print("\n Не выпадали за последние 12 игр:", end = " ")
        for temp in range(0,37):
            if(ends[temp] == 0): print(temp, end = " ")
            if(ends[temp] != 0 and temp != nom and count > 10):
                ends[temp] -= 1
        print("\nКрасное/черное", end = " ")
        print(red, black)
        count+=1
    

File: Practice/17/c++/casino/casino/casino.cpp
#include <iostream>

using namespace std;

int main(){
	setlocale(LC_ALL, "rus");
	int i = 1;
	int max = 0;
	int red = 0;
	int black = 0;
	int nom;
	int noms[37];
	for (int i = 0; i < 37; i++) {
		noms[i] = 0;
	}
	int ends[37];
	for (int i = 0; i < 37; i++) {
		ends[i] = 0;
	}
	int count = 0;
	while (i) {
		cout << "Введите последний результат\n";
		cin >> nom;
		if (nom == -1) break;
		if (nom >= 0 && nom < 37) {
			noms[nom]++;
			ends[nom] = 1 + count%12;
			if (nom % 2 == 0 && nom != 0) black++;
			if (nom % 2 == 1 && nom != 0) red++;
			for (int i = 0; i < 37; i++) {
				if (noms[i] > max) max = noms[i];
			}
			cout << "Выпадали чаще всего: ";
			for (int i = 0; i < 37; i++) {
				if (noms[i] == max) cout << i << " ";
			}
			cout << "\nНе выпадали за последние 12 игр: ";
			for (int i = 0; i < 37; i++) {
				if(ends[i] == 0) cout << i << " ";
				if (ends[i] != 0 && i != nom && count > 10) ends[i]--;
			}
			cout << "\nКрасное/черное ";
			cout << red << " " << black << "\n";
			count++;
		}
	}
}

